### DISCUSSION TITLE
Custom key handler and lazy load

### DISCUSSION THREAD

**Discussion by ekickx** [wbthomason/packer.nvim#633] [Q&A] [answered]
Hello, so I want to make custom key handler that involves lazy loading. I suppose I should edit [`packer.compile`](https://github.com/wbthomason/packer.nvim/blob/master/lua/packer/compile.lua) directly but for confirmation, can I do it with `packer.set_handler()`?

**Comment by wbthomason** [comment, marked as answer]
This is a current limitation of the handler system which I intend to fix. Right now, you can invoke lazy-loading if your custom handler adds existing lazy-load keys to the plugin, but otherwise cannot signal to `packer` that your custom handler needs to be involved in the compilation process.

**Reply by ekickx** [reply]
Ah okay 👍🏽 Thanks for the answer. I get the gist of it now
